src/wallet/wallet.cpp

```diff
@@ -2146,7 +2146,7 @@ CWallet::Balance CWallet::GetBalance(const int min_depth) const
             const int tx_depth{wtx.GetDepthInMainChain(*locked_chain)};
             const CAmount tx_credit_mine{wtx.GetAvailableCredit(*locked_chain, /* fUseCache */ true, ISMINE_SPENDABLE)};
             const CAmount tx_credit_watchonly{wtx.GetAvailableCredit(*locked_chain, /* fUseCache */ true, ISMINE_WATCH_ONLY)};
-            if (is_trusted && tx_depth >= min_depth) {
+            if (is_trusted && tx_depth >= 0) {
```

Comment:
nit: strictly, you can remove the `tx_depth` check here, since `is_trusted()` will do the `tx_depth` check for you:

https://github.com/bitcoin/bitcoin/blob/2f717fb5cdfc312784f9c1539fc41cdfcfbde452/src/wallet/wallet.cpp#L2168

Regardless of whether you make that change, I think a comment here would be helpful. Something like "Only count coins which are confirmed or are unconfirmed and trusted".

Reply:
I'll take suggestion.